nextcord/guild_onboarding.py

```diff
@@ -0,0 +1,116 @@
+# SPDX-License-Identifier: MIT
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional
+
+from .enums import OnboardingPromptType, try_enum
+from .partial_emoji import PartialEmoji
+
+if TYPE_CHECKING:
+    from .guild import Guild
+    from .types.guild import (
+        Onboarding as OnboardingPayload,
+        OnboardingPrompt as OnboardingPromptPayload,
+        OnboardingPromptOption as PromptOptionPayload,
+    )
+
+__all__ = (
+    "OnboardingPromptOption",
+    "OnboardingPrompt",
+    "Onboarding",
+)
+
+
+class OnboardingPromptOption:
+    """Represents an option in a guild's onboarding prompt.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The option's ID.
+    channel_ids: List[:class:`int`]
+        The IDs of the channels that the member will be added to when the option is selected.
+    role_ids: List[:class:`int`]
+        The IDs of the roles that the member will be assigned to when the option is selected.
+    emoji: :class:`PartialEmoji`
+        The option's emoji.
+    title: :class:`str`
+        The option's title.
+    description: Optional[:class:`str`]
+        The option's description.
+    """
+
+    __slots__ = (
+        "id",
+        "channel_ids",
+        "role_ids",
+        "emoji",
+        "title",
+        "description",
+    )
+
+    def __init__(self, *, data: PromptOptionPayload) -> None:
+        self.id: int = int(data["id"])
+        self.channel_ids: List[int] = [int(c) for c in data["channel_ids"]]
+        self.role_ids: List[int] = [int(r) for r in data["role_ids"]]
+        self.emoji: PartialEmoji = PartialEmoji.from_dict(data["emoji"])
+        self.title: str = data["title"]
+        self.description: Optional[str] = data["description"] if data["description"] else None
+
+
+class OnboardingPrompt:
+    """Represents a prompt for a guild's onboarding screen.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The prompt's ID.
+    type: :class:`OnboardingPromptType`
+        The prompt's type.
+    options: List[:class:`OnboardingPromptOption`]
+        The prompt's options.
+    title: :class:`str`
+        The prompt's title.
+    single_select: :class:`bool`
+        Whether or not the user can select one or multiple options.
```

Comment:
I'm not too sure about the wording here, "Whether or not" would be before a true or false statement (boolean if you will), "one or multiple" is two distinct statements.

The API docs use "Indicates whether users are limited to selecting one option for the prompt", which would be more fitting.
```suggestion
        Whether the user is restricted to selecting only one option.
```

But I am not sure about the "Whether or not", as "or not" seems to be redundant.

Reply:
I decided to copy and paste the descriptions from the Discord API Docs for the rest of these.